staging/src/k8s.io/kubectl/pkg/explain/v2/explain_test.go

```diff
@@ -154,31 +84,71 @@ func TestExplainErrors(t *testing.T) {
 func TestExplainOpenAPIClient(t *testing.T) {
 	var buf bytes.Buffer
 
-	fakeClient := &FakeOpenAPIV3Client{
-		Values: map[string]map[string][]byte{
-			"apis/discovery.k8s.io/v1": {
-				runtime.ContentTypeJSON: []byte(apiDiscoveryJSON),
-			},
-		},
-	}
+	fileClient := openapitest.NewFileClient(openAPIV3SpecDir)
+	paths, err := fileClient.Paths()
+	require.NoError(t, err)
+	gv, found := paths[apiGroupsPath]
+	require.True(t, found)
+	discoveryBytes, err := gv.Schema("application/json")
+	require.NoError(t, err)
+
+	var doc map[string]interface{}
+	err = json.Unmarshal(discoveryBytes, &doc)
+	require.NoError(t, err)
 
 	gen := NewGenerator()
-	err := gen.AddTemplate("Context", "{{ toJson . }}")
+	err = gen.AddTemplate("Context", "{{ toJson . }}")
 	require.NoError(t, err)
 
 	expectedContext := TemplateContext{
-		Document:  apiGroupsDocument,
+		Document:  doc,
 		GVR:       apiGroupsGVR,
 		Recursive: false,
 		FieldPath: nil,
 	}
 
-	err = printModelDescriptionWithGenerator(gen, nil, &buf, fakeClient, apiGroupsGVR, false, "Context")
+	err = printModelDescriptionWithGenerator(gen, nil, &buf, fileClient, apiGroupsGVR, false, "Context")
 	require.NoError(t, err)
 
 	var actualContext TemplateContext
 	err = json.Unmarshal(buf.Bytes(), &actualContext)
 	require.NoError(t, err)
 	require.Equal(t, expectedContext, actualContext)
-	require.Equal(t, fakeClient.FetchCounts["apis/discovery.k8s.io/v1"][runtime.ContentTypeJSON], 1)
+}
+
+// forceErrorClient always returns an error for "Paths()".
+type forceErrorClient struct{}
```

Comment:
These are simple enough, thanks. We might consider moving them to `openapitest` if they are used in multiple places, but having them here for now sounds good. 👍